gnomad/utils/transcript_annotation.py

```diff
@@ -416,10 +489,152 @@ def perform_tx_annotation_pipeline(
+        +----------+---------+---------------+---------+---------+
+        | chr1:4   | gene2   | 0.7           | 0.5     | 0.6     |
+        +----------+---------+---------------+---------+---------+
+        | chr1:5   | gene2   | 0.7           | 0.5     | 0.6     |
+        +----------+---------+---------------+---------+---------+
+        | chr1:6   | gene2   | 0.8           | 0.6     | 0.7     |
+        +----------+---------+---------------+---------+---------+
+
+    .. table:: Output Hail Table
+        :widths: auto
+
+        +---------+--------------------------------------------------------------+
+        | gene_id | regions                                                      |
+        +=========+==============================================================+
+        | gene1   | [{'chrom': 'chr1', 'start': 1, 'stop': 2, 'mean': 0.5,       |
+        |         | 'tissues': {'tissue1': 0.2, 'tissue2': 0.3}},                |
+        |         | {'chrom': 'chr1', 'start': 3, 'stop': 3, 'mean': 0.6,        |
+        |         | 'tissues': {'tissue1': 0.3, 'tissue2': 0.4}}]                |
+        +---------+--------------------------------------------------------------+
+        | gene2   | [{'chrom': 'chr1', 'start': 4, 'stop': 5, 'mean': 0.7,       |
+        |         | 'tissues': {'tissue1': 0.5, 'tissue2': 0.6}},                |
+        |         | {'chrom': 'chr1', 'start': 6, 'stop': 6, 'mean': 0.8,        |
+        |         | 'tissues': {'tissue1': 0.6, 'tissue2': 0.7}}                 |
+        +---------+--------------------------------------------------------------+
+
+    :param ht: Input Hail Table with transcript expression information.
+    :return: Hail Table with transcript annotations by region.
+    """
+    # Get a list of tissues and drop the 'tissues' field from the Table.
+    tissues = hl.eval(ht.tissues)
+    ht = ht.drop("tissues")
+
+    # Slightly restructure fields and replace NaNs and missing values with 0s.
+    set_nan_to_zero = lambda x: hl.if_else(hl.is_missing(x) | hl.is_nan(x), 0.0, x)
+    ht = ht.select(
+        "gene_id",
+        chrom=ht.locus.contig,
+        pos=ht.locus.position,
+        mean=set_nan_to_zero(ht.exp_prop_mean),
+        tissues=hl.struct(**{t: set_nan_to_zero(ht[t]) for t in tissues}),
+    )
+
+    # Order by gene_id and position, then drop the locus field.
+    ht = ht.order_by(ht.gene_id, hl.asc(ht.pos)).drop("locus")
+
+    # Key by all fields except 'pos' and collect by key into a field named 'pos'.
+    ht = ht.key_by(*[r for r in ht.row_value if r != "pos"]).collect_by_key("pos")
+
+    # Annotate with 'start' and 'stop' positions for regions by merging adjacent
+    # positions.
+    ht = ht.annotate(
+        pos=hl.fold(
+            lambda i, j: hl.if_else(
+                j.pos > (i[-1][1] + 1),
+                i.append((j.pos, j.pos)),
+                i[:-1].append((i[-1][0], j.pos)),
```

Comment:
does this line exist just to make sure the if and else clauses both return Tuples? this line doesn't do anything, right?

Reply:
I think it does do something though. It's like this example in regular python 
```
pos = [6, 7, 8, 10, 11, 12]
start = [(pos[0], pos[0])]
print("start:", start)

regions = start[:]
for current_pos in pos[1:]:
    if current_pos > (regions[-1][1] + 1):
        regions.append((current_pos, current_pos))
    else:
        last_region_first_position = regions[-1][0]
        regions = regions[:-1]
        regions.append((last_region_first_position, current_pos))

print("Final regions:", regions)
```

It prints this:
```
start: [(6, 6)]
Final regions: [(6, 8), (10, 12)]
```

Reply:
ohh thank you for this example! I must have done something weird in my testing, since it looked like it wasn't appending anything